### PR TITLE
Updates to azure-core-test

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

- Removes the `test` scope from some dependencies so that libraries consuming azure-core-test don't need to declare them as `test` dependencies in their POMs.
- Migrated serialization to `azure-json`.
- Switched out `commons-compress` for `ant` as it has fewer transitive dependencies, and those it has are less commonly used.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/core/azure-core-test/pom.xml [hunk 1]
```diff
@@ -74,6 +74,9 @@
       <artifactId>azure-core</artifactId>
       <version>1.52.0-beta.1</version> <!-- {x-version-update;com.azure:azure-core;current} -->
     </dependency>
+
+    <!-- Azure HTTP client implementation depdencnies -->
```
Comment (joshfree) [inline sdk/core/azure-core-test/pom.xml:78]: fix typo: dependencies

File: sdk/core/azure-core-test/pom.xml [hunk 2]
```diff
@@ -95,34 +98,37 @@
...
+    </dependency>
     <dependency>
       <groupId>org.junit.jupiter</groupId>
       <artifactId>junit-jupiter-params</artifactId>
       <version>5.9.3</version> <!-- {x-version-update;org.junit.jupiter:junit-jupiter-params;external_dependency} -->
     </dependency>
 
+    <!-- Used to unpack the tar.gz files of Test Proxy server -->
+    <!-- This used to be org.apache.commons:commons-compress but it was replaced with org.apache.ant:ant -->
+    <!-- This was done as ant has fewer transitive dependencies, meaning there is less chance of conflicts with other libraries. -->
```
Comment (joshfree) [inline sdk/core/azure-core-test/pom.xml:127]: please add an entry to https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-test/CHANGELOG.md

**Review by joshfree — APPROVED** [review]

File: sdk/core/azure-core-test/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-test/pom.xml:78]: Will do in a follow-up PR alongside the azure-json update

File: sdk/core/azure-core-test/pom.xml (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-test/pom.xml:127]: Will do in a follow-up PR alongside the azure-json update

File: sdk/core/azure-core-test/pom.xml (hunk 1 above)
Comment (samvaity) [inline sdk/core/azure-core-test/pom.xml:78]:
```suggestion
    <!-- Azure HTTP client implementation dependencies -->
```

File: sdk/core/azure-core-test/pom.xml [hunk 3]
```diff
@@ -95,34 +98,37 @@
...
+    <!-- Downstream libraries no longer need to add test dependencies on these libraries. -->
     <dependency>
       <groupId>io.projectreactor</groupId>
       <artifactId>reactor-test</artifactId>
       <version>3.4.38</version> <!-- {x-version-update;io.projectreactor:reactor-test;external_dependency} -->
     </dependency>
 
-    <dependency>
-      <groupId>org.apache.commons</groupId>
-      <artifactId>commons-compress</artifactId>
```
Comment (samvaity) [inline sdk/core/azure-core-test/pom.xml:106]: Is this good to be removed from external_dependency list then?

File: sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java [hunk 4]
```diff
@@ -356,8 +354,9 @@ public void close() {
             if (testProxyEnabled) {
                 testProxyRecordPolicy.stopRecording(proxyVariableQueue);
             } else {
-                try (BufferedWriter writer = Files.newBufferedWriter(createRecordFile(playbackRecordName).toPath())) {
-                    RECORD_MAPPER.writeValue(writer, recordedData);
+                try (BufferedWriter writer = Files.newBufferedWriter(createRecordFile(playbackRecordName).toPath());
```
Comment (samvaity) [inline sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java:357]: I think in longer run we should be good to make a breaking change and remove older APIs before test proxy. Now that we have the migration done.

**Pull request merged**